test/unit/components/navpatientheader/index.test.js

```diff
@@ -0,0 +1,298 @@
+
+const expect = chai.expect;
+
+const defaultPatientProps = { 
+  userid: 1234,
+  profile: { 
+    fullName: 'Vasyl Lomachenko',
+    patient: { birthday: '2010-10-20', mrn: '567890' }
+  }
+};
+
+describe('NavPatientHeader', () => {
+  const mockHistory = { push: sinon.stub() }
+  const mockTrackMetric = sinon.stub();
+
+  NavPatientHeader.__Rewire__('useHistory', sinon.stub().returns(mockHistory));
+
+  afterEach(() => {
+    mockTrackMetric.reset();
+    mockHistory.push.reset();
+  });
+
+  describe('visibility of patient info and actions', () => {
+    context('personal user with non-root permissions', () => {
+      const props = { // personal user with non-root permissions;
+        user: { roles: [] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: false } },
+      };
+
+      const wrapper = mount(
+        <BrowserRouter>
+          <NavPatientHeader {...props} />
+        </BrowserRouter>
+      );
+
+      it('shows the correct info and actions', () => {
+        expect(wrapper.text()).to.include('Vasyl Lomachenko');
+        expect(wrapper.text()).not.to.include('567890');
+        expect(wrapper.text()).not.to.include('2010-10-20');
+
+        expect(wrapper.find('button#navPatientHeader_backButton').exists()).to.be.false;
+        expect(wrapper.find('button#navPatientHeader_viewDataButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_profileButton').exists()).to.be.true;
+
+        expect(wrapper.find('button#navPatientHeader_shareButton').exists()).to.be.false;
+        expect(wrapper.find('button#navPatientHeader_uploadButton').exists()).to.be.false;
+      })
+    });
+
+    context('personal user with root permissions', () => {
+      const props = { // personal user with root permissions;
+        user: { roles: [] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: true } },
+      };
+
+      const wrapper = mount(
+        <BrowserRouter>
+          <NavPatientHeader {...props} />
+        </BrowserRouter>
+      );
+
+      it('shows the correct info and actions', () => {
+        expect(wrapper.text()).to.include('Vasyl Lomachenko');
+        expect(wrapper.text()).not.to.include('567890');
+        expect(wrapper.text()).not.to.include('2010-10-20');
+
+        expect(wrapper.find('button#navPatientHeader_backButton').exists()).to.be.false;
+        expect(wrapper.find('button#navPatientHeader_viewDataButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_profileButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_shareButton').exists()).to.be.true;
+        expect(wrapper.find('button#navPatientHeader_uploadButton').exists()).to.be.true;
+      })
+    });
+
+    context('clinician user with non-root permissions', () => {
+      const props = { // personal user with root permissions;
```

Comment:
Might need to update these comments

Reply:
copypasta strikes again

fd61aa8ca714e191a3ae29f6acfce310111e8855